src/Language/PureScript/PSString.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Language.PureScript.PSString (PSString, mkString, renderJSON, toUTF16CodeUnits, codePoints) where
+
+import Prelude.Compat
+import Numeric (showHex)
+import Data.List (unfoldr)
+import Data.Monoid ((<>))
+import Data.String (IsString(..))
+import Data.Text (Text)
+import Data.Word (Word16)
+import qualified Data.Aeson as A
+import qualified Data.Text as T
+
+-- |
+-- Strings in PureScript are sequences of UTF-16 code units, which do not
+-- necessarily represent UTF-16 encoded text. For example, it is permissible
+-- for a string to contain *lone surrogates,* i.e. characters in the range
+-- U+D800 to U+DFFF which do not appear as a part of a surrogate pair.
+--
+--
+newtype PSString = PSString [Word16]
+  deriving (Eq, Ord, A.FromJSON, Monoid)
+
+instance Show PSString where
+  show = show . codePoints
```

Comment:
Ah, I see. My response would be please let's not use `Show` for either of those things. We used to do that for types like `Ident` and it caused a few bugs - I remember a conscious effort to remove all uses of `show` for that kind of thing a while back. I think the `Show` instances for types we define in the compiler like this one should only be there so that we can work with things easily in the REPL (perhaps @paf31 or @garyb could like confirm that this is the approach we want to take?)